### PR TITLE
Add warning for mapped domain with wrong NS config

### LINKED ISSUE #3152: Domains: Show warning for mapped domains when NS records are not set to WordPress name servers.
WP-Admin:
-
image

No warnings on Calypso

### PR DESCRIPTION
**Pull request by umurkontaci**
This fixes #3152.

Goes with D2003-code.

There's a bug where mapped domain notices don't work, fix in #6983 and D2348-code.

This adds a warning notice when you buy a mapped domain but don't update the nameserver records to point to our name servers.

There are three place where this appears:

1. My site sidebar
2. Domain Management -> Domain List
3. Domain Management -> Domain Details

@ranh or @breezyskies:  I had to change GridIcon size from 16 to 18 (It wouldn't let me commit), fyi. 

Screenshot:
-
image

Testing:
-

- Get a new mapped domain
- Confirm you see the warning on 3 places aforementioned.
- Configure your NS records to point to WordPress name servers
- Confirm the message is gone or you can compare it with an already configured domain.

PS: Although we'd like the message to visible to make people configure their site, there are some legitimate (albeit not endorsed by us) ways to get a mapped domain working without changing NS records. So there's a slight chance that seeing this message everywhere might be annoying for some.

/cc: @klimeryk @aidvu for code
/cc: @ranh @breezyskies for copy and ideas on ^

### CODE REVIEW

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 1]
```diff
@@ -51,6 +51,31 @@ export default React.createClass( {
 		return this.props.domains || [ this.props.domain ]
 	},
 
+	wrongNSMappedDomains() {
+		debug( 'Rendering wrongNSMappedDomains' );
+		const wrongMappedDomains = this.getDomains().filter( domain =>
+			domain.type === domainTypes.MAPPED && ! domain.hasZone );
+
+		let learnMoreLink = ( <a
```
Comment (stephanethomas) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: It looks like we don't need this `learnMoreLink` variable. It's only used once below.

**Comment by ranh** [comment]
The issue of configuring name servers for domains mapped to a WordPress.com site is a bit complicated, I'm not sure this notice addresses that.

There are several use cases that I think could use special attention:

* Newly mapped domain, name servers not yet configured. If name servers are not set up correctly when a domain was just added, it's not really an error, it should be seen as an integral part of the flow, like in the screenshot from wp-admin in #3152.

* Existing mapped domain that used to work correctly with wpcom name servers, but name servers suddenly changed. This could make a site unavailable so if it's done by accident it's an urgent issue, we should contact user directly about it and show a notice like "your site may be offline".

* Existing mapped domain, name servers intentionally set to point outside of WordPress.com as a way to redirect the site outside. This is normal and we shouldn't show a notice at all. Using the Domain Mapping upgrade with non-wpcom nameservers used to be the recommended way to redirect your site, before we started offering the Site Redirect upgrade. There might still be plenty of users with domain mapping upgrades set up to work like that.

---

Otherwise, the format of the notice could be a bit problematic. We can only show a very short message here, and we don't have a complete flow to address the issue, we're just pointing to a support doc from what I can tell. 

We could address this by creating a support doc dedicated to this notice, so that it would fill in for the missing flow. That could help people get the right idea even if they don't immediately understand what it means that their "name servers should be configured". 

This notice might work with that:

> Your domain "example.com" may not be set up correctly. [Learn more](#).

This doesn't really address the issues mentioned above. But it could at least make for a better notice.

@umurkontaci Let me know what you think and I can prepare a dedicated support page for this if necessary.

**Comment by umurkontaci** [comment]
> Newly mapped domain, name servers not yet configured. If name servers are not set up correctly when a domain was just added, it's not really an error, it should be seen as an integral part of the flow, like in the screenshot from wp-admin in #3152.

I agree that it's not an error. Ideally, this should be integrated in something like a post-purchase flow.

> Existing mapped domain that used to work correctly with wpcom name servers, but name servers suddenly changed. This could make a site unavailable so if it's done by accident it's an urgent issue, we should contact user directly about it and show a notice like "your site may be offline".

I think this warrants an error notice. Although I don't think there's a way for us to know whether it worked before. We can assume this category if the domain is more than X days old and still not set up.

> Existing mapped domain, name servers intentionally set to point outside of WordPress.com as a way to redirect the site outside. This is normal and we shouldn't show a notice at all. Using the Domain Mapping upgrade with non-wpcom nameservers used to be the recommended way to redirect your site, before we started offering the Site Redirect upgrade. There might still be plenty of users with domain mapping upgrades set up to work like that.

I didn't know that at all. It's impossible to differentiate this from the previous category. The only solution I can find for that is to allow people to migrate to site redirect and cancel their mapping sub; but this is way overkill for a notice. 

> We could address this by creating a support doc dedicated to this notice, so that it would fill in for the missing flow. That could help people get the right idea even if they don't immediately understand what it means that their "name servers should be configured".

I think this is the quick win here. It's relatively easy to differentiate between a new vs an old domain, so we can change the copy / flow for them in the future.

**Comment by ranh** [comment]
> I didn't know that at all. It's impossible to differentiate this from the previous category. The only solution I can find for that is to allow people to migrate to site redirect and cancel their mapping sub; but this is way overkill for a notice.

Perhaps we can start by checking how many mapped domains currently have non-wpcom name servers set up. That would give us an idea of how many people will be affected by this when the notice first launches. We have done this analysis in the past, I'll try to look up some examples.

**Comment by klimeryk** [comment]
Code looks good (althought depending on Ran's findings might need some reworking), but I just wanted to note that the variable `wrongNSMappedDomains` kinda stuck out to me (or maybe I'm weird like that), because according to our [naming conventions](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#naming-conventions) it should be `wrongNsMappedDomains`. It's used a few times and just looks... weird to me :laughing: It's hardly a pressing issue, just something to think about when (or if) this needs some reworking.

**Comment by umurkontaci** [comment]
@ranh Let me know if I can help with that.

**Comment by artpi** [comment]
Where is this name server data coming from and how often is it refreshed?
Do we worry about hitting some dns query limits?

**Comment by umurkontaci** [comment]
I'm not sure about how many resolvers we have, but DNS data is cached on every resolver depending on its TTL. NS records have usually very large TTLs but it definitely varies.

This approach was being used in wp-admin and I don't recall hitting any query limits. Those limits are pretty high, e.g. Google DNS supposedly has 500QPS, which makes it 43m/day. 

@aidvu How about going a different way, instead of checking NS records, can we `nslookup` and check if it resolves to our IPs?

I think this would get rid of the issue if the users want to keep their records at somewhere else and willing to use a subdomain with a CNAME record.

**Comment by artpi** [comment]
Splendid you're working on it!

I tested it a bit and found out:
- I have a domain that has different NS but A recording is pointing to WP - domain working, no notice (as desired, awesome!)
- I have domain that has WP NS - domain working, no notice - good!
- I have a domain that is mapped, but different NS - domain not working, notice is showing - good!

I just added "A" record to the domain with different NS, we'll see how long propagation takes and if notice will pick up on it.

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 2]
```diff
@@ -51,6 +51,31 @@ export default React.createClass( {
...
+			href="https://support.wordpress.com/domains/change-name-servers"
```
Comment (artpi) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: I am not quite sure this is a proper link...

The whole notice should be displayed, when domain is bought elsewhere and Name Servers should be set in the admin panel of domain provider.
The attached link shows how to point domain bought on wp.com to outside name servers.
I believe this is the link that was supposed to be here: https://en.support.wordpress.com/map-existing-domain/#2-ask-your-domain-provider-to-update-your-dns-settings

Also, we show the same link in case of subdomain, in which case we should give this link (I believe): https://en.support.wordpress.com/map-subdomain/#instructions-for-mapping-subdomains
And information about CNAME....

Also I really find the Domain Helper output being pretty helpful and a bit more in context then a general support page.
```
'https://en.support.wordpress.com/domain-helper/?host=' + wrongMappedDomains[0].name
```

**Comment by ranh** [comment]
After testing this we now know that app. 10% of mapped domains are not using WordPress.com name servers. Of course, we don't know if these domains are set up that way intentionally or not.

**Comment by umurkontaci** [comment]
@ranh

**Comment by umurkontaci** [comment]
@ranh What is your take from the result?

**Comment by ranh** [comment]
@umurkontaci I think 10% is a significant group of users that we don't exactly understand. Maybe we could start with a more conservative approach, for example a lower key notice that is less prominent. That said, note that we're also currently sending email reminders to owners of mapped domains that are not using out name servers, and those emails also don't address the site redirect use case.

**Comment by artpi** [comment]
@ranh :
> 10% of mapped domains are not using WordPress.com name servers. Of course, we don't know if these domains are set up that way intentionally or not

Well,  how can it even work intentionally? Something has to be wrong here.
a) the NS can be wrong
b) the domain mapping should be gone
c) User knows its wrong, but wants the mapping just in case or is afraid to delete it, or wants it for the future.

If the users are keeping the mapping "just in case", we can provide "dismiss" option for that notice, but I cannot imagine technically how keeping that mapping for the domain that points elsewhere can work.

**Comment by ranh** [comment]
I'm not sure I understand the question @artpi. [See above](https://github.com/Automattic/wp-calypso/pull/3329#issuecomment-184753001) for some of the potential use cases.

**Comment by artpi** [comment]
@ranh, my point is, that regarding your comment above:

> Newly mapped domain, name servers not yet configured

We should show the notice, because users very often forget to configure NS. There is an avalanche of tickets every day and HE remind people that that they need to update NS

> Existing mapped domain that used to work correctly with wpcom name servers, but name servers suddenly changed.

We should show the notice because obviously something is wrong

> Existing mapped domain, name servers intentionally set to point outside of WordPress.com as a way to redirect the site outside. This is normal and we shouldn't show a notice at all

I disagree. If they did that, they should delete the mapping. The mapping is essentially ineffective and **there is no indication of that anywhere**.
We should show the notice and provide maybe an option to dismiss it, but we should mark the site as inaccessible, as in this PR.

My point is: in all of these cases something is misconfigured and we should inform users of that.

**Comment by ranh** [comment]
> > Existing mapped domain, name servers intentionally set to point outside of WordPress.com as a way to redirect the site outside. This is normal and we shouldn't show a notice at all
>
> I disagree. If they did that, they should delete the mapping. The mapping is essentially ineffective and there is no indication of that anywhere.

Note that the domain mapping in this case is not ineffective. It is effectively equivalent to using the Site Redirect upgrade to redirect the site away from WordPress.com. It costs the same too. 

I guess the ideal scenario here would be that every instance of domain mapping that is used as a site redirect is converted to an actual site redirect upgrade. But that is a big project with very vague benefits.

**Comment by artpi** [comment]
> It is effectively equivalent to using the Site Redirect upgrade to redirect the site away from WordPress.com.

I am not sure about that... If my domain points to WP NS and I have a site redirect upgrade, I still can put custom settings in DNS editor:
- email (mx)
- subdomains
- some other settings
And they should work.

During support rotation I had a user that had an email in the domain that was actually mapped and then he changed NS settings, so his email stopped working and even he didnt get any information from us, because well, his email stopped working :)

I think that NS changing to point elsewhere is the most dangerous scenario of all :)

**Comment by umurkontaci** [comment]
@artpi Using mapped domains instead of Site Redirect looks like this:
You have a domain called `external.com`, which is not tied to us or WordPress at all.
You also have a `example.wordpress.com`, which is a WordPress.com site and you want to retire the WordPress.com site in favor of `external.com`. 

You have two options:
1. Buy a site redirect for `external.com`. All requests coming to `example.wordpress.com` will now redirect to `external.com`.
2. Buy a domain mapping for `external.com` and set it the primary domain. All requests coming to `example.wordpress.com` will redirect to `external.com`, since it is the primary domain.

Basically a mapped domain allows you to a set a specific domain as a primary domain and since we redirect requests to other domains to the primary domain, this can be used instead of a Site Redirect upgrade.

No DNS involved in this flow, it's just a HTTP redirect.

File: client/components/notice/style.scss [hunk 3]
```diff
@@ -41,6 +41,10 @@
 	}
 }
 
+.notice__icon-wrapper {
```
Comment (umurkontaci) [inline client/components/notice/style.scss]: @breezyskies @mikeshelton1503 I had to make a CSS hack to prevent notice icons getting smaller as the number of lines are getting higher. Is there proper solution to this? I think this is a regression.

**Comment by umurkontaci** [comment]
I changed the notice level to a warning instead of an error.
I also added a different messaging for subdomain mapping, which are handled by CNAME records instead of NS records.

image

**Comment by umurkontaci** [comment]
> I just added "A" record to the domain ( 3 ) with different NS, we'll see how long propagation takes and if notice will pick up on it.

Having A records pointing to WordPress.com is not supported. It might work, but it can stop without notice should the IPs change.


> Mapping subdomain, different NS, via CNAME - subdomain not working yet (propagating) - notice shows error (as expected)

I tried CNAME and have no error for this. What domain was this?



> Would be cool to list offending domains.

I like this, we can certainly do it.


> Here it would be cool to mark the actual domain in the card with domain (like Primary mark)

I like this as well but given the required work for this, I'd say this is out of scope for this PR.

**Comment by klimeryk** [comment]
Solid patch - LGTM! :+1: 
Just a reminder: we should give a heads-up to our HEs about deploying this, so that they know where/who to report if there are some tickets regarding this change.

**Comment by umurkontaci** [comment]
I got a hunch.

We are using `has_zone` property to determine whether a domain's NS records have been set to point to us. I thought this property would refer to the domain's actual NS records point to name servers, but it wasn't. A domain can have `has_zone` true for other reasons. It doesn't necessary go false once you break your config as well.

I've run some numbers on a randomly selected domains which have `has_zone` `true`, but point to non-WordPress.com servers – it's about 18% of the records.

I'm going to hold on to this PR until we provide a more reliable solution.

**Comment by BandonRandon** [comment]
I'll add this to my list of things to test. I'm wondering if we can make the error dismissable instead of forcing users to use a site-redirect. 

It's also be great if we could try to link the user directly to the support page for their registrar.

**Comment by umurkontaci** [comment]
> I'll add this to my list of things to test.

👍  Just to let you know there's a bug in this and we got carried away with some high priority issues for a while. I hope to get back to this soon.

> I'm wondering if we can make the error dismissable instead of forcing users to use a site-redirect.

I think there's nothing wrong with making users use site redirect if that's what they want. As far as I remember, the numbers didn't look scary so I don't want to over engineer the solution. If it creates too much requests for something we didn't foresee, we can then look alternative solutions.

> It's also be great if we could try to link the user directly to the support page for their registrar.

I disagree. We can't keep an up to date list of the correct URLs for support pages of the other registrars inside Calypso and I don't want to have a broken link inside Calypso. Much better to have a support link.

We also don't know much about their registrar at this point and it needs WHOIS parsing to get this data. WHOIS parsing is not an easy task since every registry have it's own format.

**Comment by spncrb** [comment]
I agree re: site redirect - if that is what a user wants we should have them use the right upgrade. Even though this might generate some extra tickets it 1) gets us to update their upgrades to match their purpose, a net gain of information and 2)I can't imagine we will get more interactions about site redirects than we have now about how to update name servers

I think this PR is a good first step toward giving the user better information to self-fix their mapped domains.

Right now, we only tell them once on the thank you page, and then in emails sporadically over the first 21 days.  This PR adds a persistent notice that shows them next steps and while I think eventually we could build better information for them (like a custom page that shows them their registrar, what their current name servers are, what they should be replaced with, etc), this first step is moving us in the right direction.

**Comment by umurkontaci** [comment]
This is ready for a review!! @klimeryk @aidvu @spncrb please feel free to grab D2003-code and test it :)

**Comment by klimeryk** [comment]
@umurkontaci: are you sure you've pushed all the updates? From what I can see, this code relies only on `has_zone` (which you deemed not worthy [above](https://github.com/Automattic/wp-calypso/pull/3329#issuecomment-204607254) ;)), not on the `dns_configured` property you've introduced in D2003-code. Or am I missing something here?

File: client/my-sites/current-site/index.jsx [hunk 4]
```diff
@@ -85,21 +85,21 @@ module.exports = React.createClass( {
...
-				ruleWhiteList={ [ 'expiredDomains', 'expiringDomains' ] } />
+				ruleWhiteList={ [ 'expiredDomains', 'expiringDomains', 'wrongNSMappedDomains' ] } />
 		);
 	},
 
 	getSiteNotices: function() {
 		return (
 			<div>
-				{ this.getDomainExpirationNotices() }
+				{ this.getDomainWarnings() }
```
Comment (mtias) [inline client/my-sites/current-site/index.jsx]: Can you move this to the `site/notice.jsx` sub component and render it there? The design was tweaked to take less space and should be the canonical place to render site notices.

**Comment by umurkontaci** [comment]
@klimeryk I've updated the code and changed my approach a little bit, instead of checking whether the user has configured to their DNS, we're now checking whether the domain resolves to the correct IP addresses. 

This will handle users who use external nameservers and A records as well.

**Comment by mtias** [comment]
@umurkontaci SiteNotice uses a compact notice which makes it fit better within the current-site Card.

**Comment by umurkontaci** [comment]
@mtias `DomainWarnings` have its own priority rendering logic and it is being used in multiple places. It needs its own PR for investigation (#6984).

**Comment by mtias** [comment]
@umurkontaci at the very least it should be using compact notices.

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 5]
```diff
@@ -52,12 +62,20 @@ export default React.createClass( {
 	},
 
 	getPipe() {
-		const allRules = [ this.expiredDomains, this.expiringDomains, this.newDomains, this.unverifiedDomains, this.pendingGappsTosAcceptanceDomains ];
-		let rules;
+		let allRules = [
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: `allRules` should be `const`, as in the old code. We could probably simplify the code a bit by getting rid of the `rules` variable and just returning when possible.

File: client/lib/domains/assembler.js [hunk 6]
```diff
@@ -37,6 +37,8 @@ function createDomainObjects( dataTransferObject ) {
 			privateDomain: domain.private_domain,
 			registrationDate: domain.registration_date,
 			registrationMoment: domain.registration_date && i18n.moment( domain.registration_date, 'MMMM D, YYYY', 'en' ).locale( false ),
+			hasZone: domain.has_zone,
```
Comment (klimeryk) [inline client/lib/domains/assembler.js]: `hasZone` is a leftover from the old approach, right?

File: client/lib/domains/test/assembler.js [hunk 7]
```diff
@@ -48,7 +48,9 @@ describe( 'assembler', () => {
 			privateDomain: undefined,
 			registrationDate: undefined,
 			registrationMoment: undefined,
-			type: domainTypes.SITE_REDIRECT
+			type: domainTypes.SITE_REDIRECT,
+			hasZone: undefined,
```
Comment (klimeryk) [inline client/lib/domains/test/assembler.js]: Same here - `hasZone` is unused in the current approach.

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 8]
```diff
@@ -68,6 +86,58 @@ export default React.createClass( {
...
+			} else {
+				text = this.translate( '%(domainName)s\'s name server records should be configured', {
+					args: { domainName: domain.name },
+					context: 'Notice for mapped domain notice with NS records pointing to somewhere else'
+				} );
+				learnMoreUrl = support.DOMAIN_HELPER_PREFIX + domain.name;
+			}
+		} else {
+			offendingList = <ul>{ wrongMappedDomains.map( domain => <li key={ domain.name }>{ domain.name }</li> ) }</ul>;
+			if ( every( map( wrongMappedDomains, 'name' ), isSubdomain ) ) {
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Nice! 👍

File: client/my-sites/upgrades/components/domain-warnings/style.scss [hunk 9]
```diff
@@ -1,4 +1,4 @@
-.domain-warnings__unverified-domains {
+.domain-warnings-notice {
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/style.scss]: I don't want to be a PITA, but since you're already changing the name of this CSS class, might as well change it to something that doesn't generate an eslint warning. According to our [CSS guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md), this should be `domain-warnings__notice`.

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 10]
```diff
@@ -222,11 +310,16 @@ export default React.createClass( {
...
+				isCompact={ this.props.isCompact }
+				key="pending-gapps-tos-notice"
+				siteSlug={ this.props.selectedSite && this.props.selectedSite.slug }
+				domains={ pendingDomains }
+				section="domain-management" />;
 	},
 
 	componentWillMount: function() {
-		if ( ! this.props.domains || ! this.props.domain ) {
+		if ( ! this.props.domains && ! this.props.domain ) {
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Whoa, nice catch! 👍

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 11]
```diff
@@ -68,6 +86,58 @@ export default React.createClass( {
...
+				} );
+				learnMoreUrl = support.MAP_EXISTING_DOMAIN_UPDATE_DNS;
+			}
+		}
+		return <Notice
+			isCompact={ this.props.isCompact }
+			status="is-warning"
+			className="domain-warnings-notice"
+			showDismiss={ false }
+			key="wrong-ns-mapped-domain">{ text } <a href={ learnMoreUrl }>{ this.translate( 'Learn more.' ) }</a>{ offendingList }</Notice>;
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: I think this link should open in a new tab - at least that was my expectation.

Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Also, can we _please_ move the dot in the sentence to the left a bit? Currently it looks off to me:
screen shot 2016-08-01 at 01 19 12

It's especially stands out in the compact version:
screen shot 2016-08-01 at 01 20 13

This seems more consistent with our other warning notices:
screen shot 2016-08-01 at 01 19 53

Sorry for pointing out such minute detail, but sometimes such details are what make the UI seem more polished and professional. Also, you know I like to nitpick 😝

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 12]
```diff
@@ -68,6 +86,58 @@ export default React.createClass( {
 		return ( this.props.domains || [ this.props.domain ] ).filter( domain => domain.currentUserCanManage );
 	},
 
+	wrongNSMappedDomains() {
+		debug( 'Rendering wrongNSMappedDomains' );
+		const wrongMappedDomains = this.getDomains().filter( domain =>
+			domain.type === domainTypes.MAPPED && ! domain.pointsToWpcom );
+
+		debug( 'Domains:', this.getDomains() );
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Do we really need to print all the site's domains here? It might make more sense if we put this line somewhere "higher" in this component, maybe even in the domains list, but I still think this is a bit overkill. Also, calling `this.getDomains()` isn't exactly free, since there's some filtering involved (not that it's costly either...) and it will be called every time this line is hit (rerenders, etc), whether debugging is enabled or not (at least that's what my testing with `CALYPSO_ENV=production make build` tells me).

**Comment by klimeryk** [comment]
Code works as advertised - I've just left some minor <del>nitpicks</del> comments.

**Comment by klimeryk** [comment]
Also noting that this also fixes #2397.

File: client/my-sites/upgrades/components/domain-warnings/index.jsx (hunk 12 above)
Comment (umurkontaci) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: The line is not exactly necessary, though there won't be a measurable performance impact either way. Regardless though, I thought `debug` statements are removed in production builds.

Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Yeah, me too - but the command above produces a bundle with `debug` calls. Maybe there's a step after that that I've missed? But that should be mostly removing comments and whitespace 😕

Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Oh, and my main point here (probably lost in my rant) was that this info (the list of all the domains) is not specific to this one warning, but to all of them, or even to the the whole domains list - which is where I'd prefer to have it. Then we can benefit from it during debuging of all the different warnings, not only this exact one.

File: client/lib/domains/assembler.js (hunk 6 above)
Comment (umurkontaci) [inline client/lib/domains/assembler.js]: hasZone is included in the API response anyway. No harm in keeping it.

**Comment by umurkontaci** [comment]
Also changed the notice so it doesn't show a list of domains in SitesList since it looked super weird in compact version.

image

**Comment by umurkontaci** [comment]
Making notices compact required changing the copies of compact notices as well; it now looks like this:

image

The copies can use of a bit of adjustment, I'll hook that up in a future PR though.

**Comment by klimeryk** [comment]
Pinging @umurkontaci to resolve the conflicts on this branch before it can be reviewed (you know how it goes - sometimes a merge goes south and introduces old/bad code so a second review would be needed).

**Comment by umurkontaci** [comment]
@klimeryk Fixed the conflicts, everything seems OK.

**Comment by klimeryk** [comment]
Awesome work, @umurkontaci! 👍 
I've just made a small adjustment in https://github.com/Automattic/wp-calypso/pull/3329/commits/a8d7d991205d4beaa5affd46f0a525392233cff7 - the notice was not displayed on the domain details page (probably got lost in a merge). Other than that - looks great!
